### PR TITLE
Paths - get rid of join()

### PR DESCRIPTION
**Pull request by bpasero**
In most cases this means, paths are now having a backslash on Windows and not slash as our previous implementation.

Only for relative paths I left in the slash via `posix.join`.

### CODE REVIEW

File: src/vs/workbench/api/node/extHostLogService.ts [hunk 1]
```diff
@@ -3,15 +3,14 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { join } from 'vs/base/common/extpath';
+import { join } from 'vs/base/common/path';
```
Comment (bpasero) [inline src/vs/workbench/api/node/extHostLogService.ts]: @sandy081 fyi `getLogDirectory()` will now have backslash on Windows

File: src/vs/workbench/browser/parts/editor/breadcrumbsPicker.ts [hunk 2]
```diff
@@ -299,7 +299,7 @@ class FileFilter implements ITreeFilter<IWorkspaceFolder | IFileStat> {
 						continue;
 					}
 					let patternAbs = pattern.indexOf('**/') !== 0
-						? join(folder.uri.path, pattern)
+						? posix.join(folder.uri.path, pattern)
```
Comment (bpasero) [inline src/vs/workbench/browser/parts/editor/breadcrumbsPicker.ts]: @jrieken fyi

File: src/vs/workbench/contrib/debug/node/debugAdapter.ts [hunk 3]
```diff
@@ -441,7 +440,7 @@ export class ExecutableDebugAdapter extends StreamDebugAdapter {
 		const result: IDebuggerContribution = Object.create(null);
 		if (contribution.runtime) {
 			if (contribution.runtime.indexOf('./') === 0) {	// TODO
-				result.runtime = extpath.join(extensionFolderPath, contribution.runtime);
+				result.runtime = path.join(extensionFolderPath, contribution.runtime);
```
Comment (bpasero) [inline src/vs/workbench/contrib/debug/node/debugAdapter.ts]: @weinand fyi as discussed this changes to backslash now. Please consider revisiting the TODO in the line above, maybe use `path.isAbsolute()` I tried that but it resulted in a test failure?

File: src/vs/workbench/contrib/logs/electron-browser/logs.contribution.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as nls from 'vs/nls';
-import { join } from 'vs/base/common/extpath';
+import { join } from 'vs/base/common/path';
```
Comment (bpasero) [inline src/vs/workbench/contrib/logs/electron-browser/logs.contribution.ts]: @sandy081 fyi output channels will now have backslash on Windows

File: src/vs/workbench/contrib/markers/electron-browser/markersPanel.ts [hunk 5]
```diff
@@ -288,7 +287,7 @@ export class MarkersPanel extends Panel implements IMarkerFilterController {
 		return Object.keys(expr)
 			.reduce((absExpr: IExpression, key: string) => {
 				if (expr[key] && !isAbsolute(key)) {
-					const absPattern = join(root, key);
+					const absPattern = posix.join(root, key);
```
Comment (bpasero) [inline src/vs/workbench/contrib/markers/electron-browser/markersPanel.ts]: @sandy081 fyi trying to preserve glob pattern here

File: src/vs/workbench/contrib/output/electron-browser/outputServices.ts [hunk 6]
```diff
@@ -185,7 +184,7 @@ class OutputChannelBackedByFile extends AbstractFileOutputChannel implements Out
 		@IModeService modeService: IModeService,
 		@ILogService logService: ILogService
 	) {
-		super({ ...outputChannelDescriptor, file: URI.file(extpath.join(outputDir, `${outputChannelDescriptor.id}.log`)) }, modelUri, fileService, modelService, modeService);
+		super({ ...outputChannelDescriptor, file: URI.file(join(outputDir, `${outputChannelDescriptor.id}.log`)) }, modelUri, fileService, modelService, modeService);
```
Comment (bpasero) [inline src/vs/workbench/contrib/output/electron-browser/outputServices.ts]: @sandy081 fyi output channels will now have backslash on Windows

File: src/vs/workbench/contrib/tasks/common/problemMatcher.ts [hunk 7]
```diff
@@ -188,7 +188,7 @@ export function getResource(filename: string, matcher: ProblemMatcher): URI {
 	if (kind === FileLocationKind.Absolute) {
 		fullPath = filename;
 	} else if ((kind === FileLocationKind.Relative) && matcher.filePrefix) {
-		fullPath = Extpath.join(matcher.filePrefix, filename);
+		fullPath = Path.join(matcher.filePrefix, filename);
```
Comment (bpasero) [inline src/vs/workbench/contrib/tasks/common/problemMatcher.ts]: @sandy081 fyi will now have backslash on Windows

**Review by sandy081 — APPROVED** [review]

**Comment by bpasero** [comment]
Obsolete by https://github.com/Microsoft/vscode/commit/39566fa5e348996c4a573e7901149ff8487e4af8

**Pull request closed without merging**